src/components/profile/profile.component.js

```diff
@@ -17,79 +17,73 @@ const marginPropTypes = filterStyledSystemMarginProps(
   styledSystemPropTypes.space
 );
 
-class Profile extends React.Component {
-  /** Determines whether a `src` prop has been supplied */
-  get hasSrc() {
-    return Boolean(this.props.src);
-  }
+const Profile = ({
+  src,
+  className,
+  initials,
+  name,
+  size,
+  email,
+  large,
+  ...props
+}) => {
+  const hasSrc = () => !!src;
 
-  /** Returns the classes for the component. */
-  get classes() {
-    return classNames(this.props.className);
-  }
+  const classes = () => classNames(className);
 
-  /** Returns the initials for the name. */
-  get initials() {
-    if (this.props.initials) return this.props.initials;
-    return acronymize(this.props.name);
-  }
+  const getInitials = () => {
+    if (initials) return initials;
+    return acronymize(name);
+  };
 
-  get marginProps() {
-    return filterStyledSystemMarginProps(this.props);
-  }
-
-  /** Returns the avatar portion of the profile. */
-  get avatar() {
-    if (this.hasSrc) {
+  const avatar = () => {
+    if (hasSrc()) {
       return (
         <ProfileAvatarStyle
-          src={this.props.src}
-          alt={this.initials}
-          initials={this.initials}
-          size={this.props.size}
+          src={src}
+          alt={getInitials()}
+          initials={getInitials()}
+          size={size}
           shape="square"
           data-element="user-image"
         />
       );
     }
     return (
       <ProfileAvatarStyle
-        initials={this.initials}
-        gravatar={this.props.email}
-        size={this.props.size}
+        initials={getInitials()}
+        gravatar={email}
+        size={size}
       />
     );
-  }
+  };
 
-  /** Returns the text portion of the profile. */
-  get text() {
+  const text = () => {
     return (
-      <ProfileDetailsStyle size={this.props.size} hasSrc={this.hasSrc}>
-        <ProfileNameStyle size={this.props.size} data-element="name">
-          {this.props.name}
+      <ProfileDetailsStyle size={size} hasSrc={hasSrc()}>
+        <ProfileNameStyle size={size} data-element="name">
+          {name}
         </ProfileNameStyle>
-        <ProfileEmailStyle size={this.props.size} data-element="email">
-          {this.props.email}
+        <ProfileEmailStyle size={size} data-element="email">
+          {email}
         </ProfileEmailStyle>
       </ProfileDetailsStyle>
     );
-  }
+  };
 
-  render() {
-    return (
-      <ProfileStyle
-        large={this.props.large}
-        className={this.classes}
-        hasSrc={this.hasSrc}
-        {...tagComponent("profile", this.props)}
-        {...this.marginProps}
-      >
-        {this.avatar}
-        {this.text}
-      </ProfileStyle>
-    );
-  }
-}
+  return (
+    <ProfileStyle
+      large={large}
```

Comment:
What does this prop do? It isn't used in the styled-component as far as I can tell?